torch/distributed/_tensor/examples/display_sharding_example.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any, Dict
+
+import torch
+
+from torch.distributed._tensor.debug import CommDebugMode
+
+from torch.distributed._tensor.debug.comm_mode import ModuleParamaterShardingTracker
+
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    MLPModule,
+    MLPStacked,
+)
+
+
+class DisplayShardingExample:
+    """
+    Checks if the set of keys in ground truth dictionary and the set
+    produced in advanced_module_tracker are in the same order
+    """
+
+    def same_set_of_keys(self, dict1, dict2):
+        dict1_keys = []
+        dict2_keys = []
+
+        for key in dict1:
+            for nested_key in dict1[key]:
+                dict1_keys.append((key, nested_key))
+
+        for key in dict2:
+            for nested_key in dict2[key]:
+                dict2_keys.append((key, nested_key))
+
+        if len(dict1_keys) != len(dict2_keys):
+            return False
+
+        for i in range(len(dict1_keys)):
+            if dict1_keys[i] != dict2_keys[i]:
+                return False
+
+        return True
+
+    def ground_truth(self, model):
+        module_parameters_dict: Dict[str, Any] = {}
+
+        for name, parameters in model.named_parameters():
+            module_name = model.__class__.__name__ + "." + name.rsplit(".", 1)[0]
+            parameter_name = name.rsplit(".", 1)[1]
```

Comment:
Could you add comments on what the split does here? It is not easy to reason from what it looks like.